PlanMyTrip/PlanMyTrip/View Controllers/HotelRequirementsViewController.m

```diff
@@ -22,68 +22,46 @@ - (IBAction)saveData:(id)sender;
-        [request2 setHTTPMethod:@"GET"];
-        [request2 setAllHTTPHeaderFields:headers2];
-
-        NSURLSession *session2 = [NSURLSession sharedSession];
-        NSURLSessionDataTask *dataTask2 = [session2 dataTaskWithRequest:request2
-                                                    completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
-                                                        if (error) {
-                                                            NSLog(@"%@", error);
-                                                        } else {
-                                                            NSMutableArray *res = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
-                                                            NSMutableDictionary *item = [res firstObject];
-                                                            self.flightInfoSaved2.arrivalCity = [item objectForKey:@"id"];
-                                                        }
-                                                    }];
-        [dataTask2 resume];
-    }
 }
 
+-(void)callAPI:(NSString *)location action:(id)sender{
+    NSDictionary *headers = @{ @"X-RapidAPI-Key": @"c151066f31mshf429fe6db920209p199187jsnaf96ca6dffe5",
+                               @"X-RapidAPI-Host": @"priceline-com-provider.p.rapidapi.com" };
+    
+    NSString *urlT = [NSString stringWithFormat:@"https://priceline-com-provider.p.rapidapi.com/v1/hotels/locations?name=%@&search_type=ALL", location];
+    
+    NSString* url = [urlT stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
+    
+    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:url] cachePolicy:NSURLRequestUseProtocolCachePolicy timeoutInterval:10.0];
+    [request setHTTPMethod:@"GET"];
+    [request setAllHTTPHeaderFields:headers];
+    NSURLSession *session = [NSURLSession sharedSession];
+    NSURLSessionDataTask *dataTask = [session dataTaskWithRequest:request
+                                                completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+                                                    if (error) {
+                                                        NSLog(@"%@", error);
+                                                    } else {
+                                                        NSMutableArray *resDict = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
+                                                        NSMutableDictionary *firstElem = [resDict firstObject];
+                                                        dispatch_async(dispatch_get_main_queue(), ^{
+                                                            if(self.carStatus == 1){
+                                                                self.destination.text = [firstElem objectForKey:@"id"];
```

Comment:
Isn't this line the same for both cases? then you can leave it outside of the if-else.